app/assets/javascripts/tylium/modules/auto_save/local.js

```diff
@@ -0,0 +1,144 @@
+class LocalAutoSave {
+  constructor(target) {
+    if (target.tagName !== 'FORM') { console.error('Can\'t initialize local auto save on anything but a form'); return; }
+    this.target = target;
+    this.key = target.dataset.autoSaveKey;
+    this.cancelled = false;
+
+    // List of available inputs: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input
+    // Only permit these inputs to be saved so that it does not store unnecessary data in local cache
+    this.permittedInputTypes = [
+      'checkbox',
+      'color',
+      'date',
+      'email',
+      'hidden', // Needed for hidden tag_input in issues/form
+      'number',
+      'radio',
+      'tel',
+      'text'
+    ];
+
+    // Don't store authenticity_token and utf8
+    this.excludedInputNames = ['utf8', 'authenticity_token', '_method'];
+
+    this.init();
+  }
+
+  init() {
+    this.behaviors();
+    this.restoreData();
+  }
+
+  behaviors() {
+    var that = this;
+
+    this.target.addEventListener('submit', function(event) {
+      localStorage.removeItem(that.key);
+    });
+
+    var clearCacheElement = this.target.querySelector('[data-behavior~=clear-local-auto-save]');
+
+    if (clearCacheElement) {
+      clearCacheElement.addEventListener('click', function(event) {
+        that.cancelled = true;
+        localStorage.removeItem(that.key);
+      })
+    }
+
+    // Find all inputs in the form, then exclude based on excluded input types
+    var formInputs = Array.from(this.target.querySelectorAll('input')).filter(function(input) {
+      return that.permittedInputTypes.includes(input.getAttribute('type')) && !that.excludedInputNames.includes(input.name);
+    })
+
+    formInputs = formInputs.concat(Array.from(this.target.querySelectorAll('textarea, select')));
+
+    formInputs.forEach(function(input) {
+      // we're using a jQuery plugin for :textchange event, so need to use $()
+      $(input).on('textchange change', that.debounce(that.setData.bind(that), 500));
+    })
+  }
+
+  debounce(func, delay) {
```

Comment:
@brianp This worked perfectly! Thanks

Reply:
@sean-yeoh I think I see why it might be failing sometimes. I've double used a variable in this example and shouldn't have. 

It should read:

```javascript
 handleTextChange() {
    clearTimeout(this._debounceTimeout);

    this._debounceTimeout = setTimeout(function() {
      this.setData();
    }.bind(this), this._debounceTimer);
  }
```

`this._debounceTimer` should represent the timing interval and `this._debounceTimeout` should be the timeout. One was clobbering the other.

Reply:
@sean-yeoh This will still be an issue. We need to separate the `debounceTimeout` and the `debounceTimer` otherwise we're overwriting the wrong variables.